pkg/agent/tunnelserver/stream.go

```diff
@@ -46,18 +65,62 @@ type streamWriter struct {
 	lastMessage  time.Time
 	bytesWritten int64
 	log          log.Logger
+	gitIgnore    gitignore.IgnoreParser
 }
 
 func (s *streamWriter) Write(p []byte) (int, error) {
-	err := s.stream.Send(&tunnel.Chunk{Content: p})
+	var filesToUpload []string
+	err := filepath.Walk(".", func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+		if info.IsDir() {
+			return nil
+		}
+		relPath, err := filepath.Rel(".", path)
+		if err != nil {
+			return err
+		}
+		if !strings.HasPrefix(relPath, "./") {
+			relPath = "./" + relPath
+		}
+		if !s.gitIgnore.Match(relPath, info.IsDir()) || !s.gitIgnoreEnabled { // Check if gitIgnoreEnabled is true
```

Comment:
Is there a way of doing this check before we start walking the filesystem? Seems like a lot of unnecessary work if we don't want to ignore `.gitignore`